### DISCUSSION TITLE
Update MS Store apps with Winget ?

### DISCUSSION THREAD

**Discussion by m01001101-01010110** [microsoft/winget-cli#2184] [Q&A] [answered]
Hello,

Is it possible to update MS Store apps with Winget ?

I see msstore as a source for packages.

I can list them for example Firefox :
![image](https://user-images.githubusercontent.com/15172121/170073890-d68a7426-4daa-40f9-a7c9-fbb3f0ecdcf8.png)
Note that version number is unknown.

I can list Firefox as an installed app  :
![image](https://user-images.githubusercontent.com/15172121/170074349-7856fa1b-d70d-40b3-9da0-b2c0c8eeef50.png)
When I type "Winget list", apps from MS Store have an empty line for the Source column. 
Why ? Am I missing a configuration or something ?

Thank you.

**Comment by denelon** [comment, marked as answer]
Yes, you can run `winget upgrade <ID>` or `winget upgrade "Package name"`.

There are a couple of things going on here.

The first as you called out is the "msstore" source doesn't provide package versions yet. If you run `winget upgrade --all` you may see a count of packages with unknown versions. The store will be adding versions in the future to improve the upgrade experience.

The second is due to some correlation improvements we have been working on to match installed packages with manifests in configured sources. This will improve over time.

No, you aren't missing any configuration. As we continue to improve the correlation logic, we will be able to continue improving `winget list` to display more packages from the Microsoft Store.

**Reply by treysis** [reply]
This is not working (anymore?).

**Comment by wolf99** [comment]
Side note: I hope that if msstore manifest is adding version information, then we may  be able to see information on the currently released version in the store, rather than only the curently installed version...

**Reply by denelon** [reply]
Unfortunately, this is a non-trivial problem to solve. We were able to get version information for store packages via the endpoint where we call "show" for some of the packages in the Microsoft Store's REST endpoint. We're hoping the challenges can be solved to provide the information needed in the other endpoints in the interface so WinGet can see which version(s) of store packages are available. Additional work is being done to pass the additional information from WinGet to the store endpoint so the appropriate version information can be displayed.
